File: Entities/Models/Book.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Models
{
    public class Book
    {
        public int Id { get; set; }
        public int AuthorId { get; set; }

        public string? Title { get; set; }
        public string? ProductImageName { get; set; }
        public string? Description { get; set; }
        public string? DocxFileName { get; set; }

        public int Likes { get; set; }
        public int Views { get; set; }
        public int Favorits { get; set; }
        public decimal Price { get; set; }

        public Author? Author { get; }
        public ICollection<Genre>? Genres { get; set; }
    }
}

File: blibt.client/src/app/shared/models/Book.ts
import { Genre } from "./Genre";

export class Book {
  id: number = 0;
  authorId: number=0;
  title: string = '';
  productImageName: string = '';
  genresIds: Array<string> = [];;
  description: string = '';
  likes: number = 0;
  views: number = 0;
  favorits: number = 0;
  price: number = 0
  ImageFile?: File;


  genres: Array<Genre> = [];
  authorName: string = '';

}
